### PR TITLE
Add content and page types.

### PR DESCRIPTION
**Pull request by stbof**
https://cockroachlabs.atlassian.net/browse/DOC-1765

### CODE REVIEW

File: StyleGuide.md [hunk 1]
```diff
@@ -152,91 +174,158 @@ Examples:
...
-- [JSON Support](https://www.cockroachlabs.com/docs/stable/demo-json-support.html)
-- [Spatial tutorial](https://www.cockroachlabs.com/docs/stable/spatial-tutorial.html)
+- [Deploy a Local Cluster with Kubernetes](https://www.cockroachlabs.com/docs/stable/orchestrate-a-local-cluster-with-kubernetes.html)
+- [Visualize CockroachDB Schemas with DBeaver](https://www.cockroachlabs.com/docs/stable/dbeaver.html)
 
-### Reference docs
+### Best practice
 
-Reference docs provide information about a specific CockroachDB function, feature, or interface. Reference docs are detail-oriented, and should include all of the information available on a specific topic, without providing prescriptive guidance.
+A _best practice_ guide is a set of recommendations on how to choose among CockroachDB features to achieve specific goals. Goals can include throughput, latency, and survivability.
```
Comment (mikeCRL) [inline StyleGuide.md:285]:
```suggestion
A _best practice_ guide is a set of recommendations on how to choose among CockroachDB features and their available configurations to achieve specific goals. Goals can include throughput, latency, survivability, and security.
```

**Review by mikeCRL — COMMENTED** [review]
Left some comments. I appreciate the additional clarity and specificity here. The relationship between content types and page types - the nuance that some types can exist at only one level, others at multiple levels - may be hard for a new reader to understand in the current structure. One idea that came to mind (in case it resonates with you) is a matrix where all types are listed and columns allow us to specify where in the docs these might be used (i.e. one or more contexts; page vs. heading).

I also left some specific comments about what's allowed or disallowed at each level per this PR.

Many nice, unrelated edits here, as well. Thank you.

I was speaking with @jseldess about this work again recently and I think he may have additional feedback and ideas to share.

File: StyleGuide.md [hunk 2]
```diff
@@ -152,91 +174,158 @@ Examples:
...
+A page or a heading within a page can be one of these content types.
 
-When writing a document, you should follow these type definitions with a degree of flexibility, as there is some amount of cross-over between them.
+### Concept
 
-### How-to docs
+A _concept_ explains how a particular feature works, or how a specific system is designed. Conceptual pages do not provide prescriptive guidance or instruction.
 
-How-to docs provide step-by-step instructions on completing a specific goal. In contrast with [tutorials](#tutorials), how-to docs are discrete, action-based, and do not need to be limited to a specific use-case. 
+- Title or heading is a plural noun.
```
[the remark this answers is not in the document]
Comment (stbof) [reply StyleGuide.md]: Fixed.

File: StyleGuide.md [hunk 3]
```diff
@@ -152,91 +174,158 @@ Examples:
...
+- [Architecture Overview](https://www.cockroachlabs.com/docs/stable/architecture/overview.html)
+- [SQL Layer](https://www.cockroachlabs.com/docs/stable/architecture/sql-layer.html)
+- [Multi-Region Capabilities Overview](https://www.cockroachlabs.com/docs/stable/multiregion-overview.html)
+
+### Task
+
+A _task_ provides step-by-step instructions to complete a specific goal. In contrast with [tutorials](#tutorials), tasks are discrete, action-based, and do not need to be limited to a specific use case.
+
+- Answers the question of "how to?" by describing precisely what to do and the order in which to do it.
+- Title or heading should state an actionable goal for the user and is of the form **\<Imperative verb\> a \<noun\>**.
```
[the remark this answers is not in the document]
Comment (stbof) [reply StyleGuide.md]: Fixed.

File: StyleGuide.md [hunk 4]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+- [Best practice](#best-practice)
+- [Troubleshooting](#troubleshooting)
+- [FAQs](#faqs)
+- [Release note](#release-note)
+
+### Tutorial
+
+A _tutorial_ is a task that provides instructions on using CockroachDB in the context of a specific use case. Tutorials acquaint users with a specific feature of CockroachDB through an end-to-end example that achieves a concrete result.
+
+A tutorial helps users quickly achieve competence in a CockroachDB feature or understand tradeoffs between different features and feature configurations. The latter usage is complementary to [Best Practice](#best-practice) guide.
```
Comment (ianjevans) [inline StyleGuide.md:323]:
```suggestion
A tutorial helps users quickly achieve competence in a CockroachDB feature or understand tradeoffs between different features and feature configurations. The latter usage is complementary to a [Best Practice](#best-practice) guide.
```

File: StyleGuide.md [hunk 5]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
-- [How-to docs](#how-to-docs)
-- [Tutorials](#tutorials)
-- [Reference docs](#reference-docs)
-- [Conceptual docs](#conceptual-docs)
+Concept content helps users understand the foundation of Cockroach Labs services and product features, including definitions of terms with specific meaning in CockroachDB and descriptions of how CockroachDB and Cockroach services are structured.
 
-When writing a document, you should follow these type definitions with a degree of flexibility, as there is some amount of cross-over between them.
+Concept content often includes a [glossary](#glossary).
+
+- The first sentence answers the question "what is?" in the form **A _\<singular noun\>_ is ….**.
```
Comment (ianjevans) [inline StyleGuide.md:211]:
```suggestion
- The first sentence answers the implicit question "what is a \<singular noun\>?" in the form **A _\<singular noun\>_ is ….**.
```

File: StyleGuide.md [hunk 6]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+
+  **Example:** An _index_ is a data structure that improves the speed of data retrieval operations on a database table at the cost of additional writes and storage space to maintain the index data structure.
+
+- For simple concept pages, the title should be a plural noun representing the concept(s) or entities(s) to be described, optionally followed by a page descriptor, such as **Overview**.
+
+  **Example:** Indexes.
+
+- Provide as much relevant information as you can, and then link to other pages as necessary.
```
Comment (ianjevans) [inline StyleGuide.md:219]:
```suggestion
- Provide as much relevant information as you can, and then link to other pages as necessary. If there is a related reference topic, link to it within the topic.
```

File: StyleGuide.md [hunk 7]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
 
-Here are some principles to follow when writing a how-to doc:
+Task content helps users efficiently develop, deploy, and manage applications.
 
-- The title of the page should state an actionable goal for the user. For example, if the goal of a user is to start a cluster, the title of the how-to doc might be "Start a Cluster".
-- The title of the page, and the steps on the page, should be second-person imperative. For example, if the first step in starting a cluster is to generate certificates, then the first header on the page might be "Generate certificates" (as opposed to "Generating certificates" or "How to generate certificates").
-- Limit the amount of reference information on the page. You can add hyperlinks to the relevant sections of separate [reference docs](#reference-docs).
+At the page level, a task could take two different forms: a page with multiple, related tasks or a single longer task.
+
+- Answers the question of "how to?" by describing precisely what to do and the order in which to do it.
```
Comment (ianjevans) [inline StyleGuide.md:237]:
```suggestion
- Answers the question of "how to do \<an action\>?" by describing precisely what to do and the order in which to do it.
```

File: StyleGuide.md [hunk 8]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+- Title or heading should state an actionable goal for the user and ideally of the form **\<Imperative verb\> [\<article\>|\<conjunction\>] \<noun\> or \<proper noun\>**.
```
Comment (ianjevans) [inline StyleGuide.md:238]:
```suggestion
- The title or heading should state an actionable goal for the user and ideally of the form **\<Imperative verb\> [\<article\>|\<conjunction\>] \<noun\> or \<proper noun\>**.
```

File: StyleGuide.md [hunk 9]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+
+  **Example:** Create an Index
+- Verb ideally should be specific. Avoid generic verbs such as **Use**, **Manage** unless naming a page containing specific tasks.
```
Comment (ianjevans) [inline StyleGuide.md:241]:
```suggestion
- The verb ideally should be specific. Avoid generic verbs such as **Use**, **Manage** unless naming a page containing specific tasks.
```

File: StyleGuide.md [hunk 10]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+- Avoid **Your** because you may be using an object that you don't "own".
+
+  **Example:** Connect to a Cluster, not Connect to Your Cluster
+- Lead with the verb. Don't bury it at the end of the heading.
+
+  **Example:** Access DB Console, not DB Console Access.
+
+- Presented as an ordered list.
```
Comment (ianjevans) [inline StyleGuide.md:249]:
```suggestion
- Present the steps as an ordered list.
```

File: StyleGuide.md [hunk 11]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+
+- The title of the page, and the steps on the page, should be second-person imperative. For example, if the first step in starting a cluster is to generate a certificate, then the first header on the page might be **Generate a certificate** (as opposed to **Generating a certificate** or **How to generate a certificate**).
+- Limit the amount of reference information on the page. You can add hyperlinks to the relevant sections of [reference](#reference) in line or in the [See also](#see-also) section.
```
Comment (ianjevans) [inline StyleGuide.md:252]:
```suggestion
- Limit the amount of reference information on the page. You can add links to the relevant sections of [reference](#reference) in line or in the [See also](#see-also) section.
```

File: StyleGuide.md [hunk 12]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
 
-Here are some principles to follow when writing a tutorial:
+Reference content helps users understand the precise meaning and affect of CockroachDB SQL language constructs, platform, configuration options, and API parameter values, etc.
 
-- Headings should include `Step N.` to denote the sequence of steps to follow.
-- Limit the number of steps. Aim for the heuristic maximum of 10 steps.
-- Make the instructions prescriptive (i.e., tell the user exactly what to do).
-- Try to keep the tutorial self-contained. It's okay to link to reference information at the end of a step, but if you are being prescriptive, the reader shouldn't have to consult multiple pages to actually complete the tutorial.
+- Should be comprehensive and accurate. This principle might apply to other pages types, but it is especially important for reference, as it is the ultimate source of truth (i.e., the "reference") for a particular feature or interface.
```
Comment (ianjevans) [inline StyleGuide.md:268]:
```suggestion
- The content should be comprehensive and accurate. This principle might apply to other pages types, but it is especially important for reference, as it is the ultimate source of truth (i.e., the "reference") for a particular feature or interface.
```

File: StyleGuide.md [hunk 13]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+- Should be succinct. Details are typically presented in tabular form. Prose is better suited for [conceptual pages](#conceptual).
```
Comment (ianjevans) [inline StyleGuide.md:269]:
```suggestion
- The content should be succinct. Details are typically presented in tabular form. Prose is better suited for [conceptual pages](#conceptual).
```

File: StyleGuide.md [hunk 14]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
 
-- [Architecture Overview](https://www.cockroachlabs.com/docs/stable/architecture/overview.html)
-- [SQL Layer](https://www.cockroachlabs.com/docs/stable/architecture/sql-layer.html)
+A _prerequisites_ section describes conditions that must be satisfied before starting a [task](#task) or [tutorial](#tutorial).
+
+### See also
+
+A _see also_ section is a list of related pages.
+
+This section is always the last section in a page.
```
Comment (ianjevans) [inline StyleGuide.md:307]:
```suggestion
This section is always the last section in a page.

Keep the list of links short, and think about whether the related pages should instead be linked within the content itself to give users more context around _why_ another topic is related. For example, if the task describes how to perform an action on the command line, and another task describes how to perform the same action in a GUI, the link to the GUI task should be in the topic's introduction, not in the See also section.

File: StyleGuide.md [hunk 15]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+- [FAQs](#faqs)
+- [Release note](#release-note)
+
+### Tutorial
+
+A _tutorial_ is a task that provides instructions on using CockroachDB in the context of a specific use case. Tutorials acquaint users with a specific feature of CockroachDB through an end-to-end example that achieves a concrete result.
+
+A tutorial helps users quickly achieve competence in a CockroachDB feature or understand tradeoffs between different features and feature configurations. The latter usage is complementary to [Best Practice](#best-practice) guide.
+
+- Title is of the form **\<Imperative verb\> xxx \<noun\>**.
```
Comment (ianjevans) [inline StyleGuide.md:325]:
```suggestion
- The title is of the form **\<Imperative verb\> xxx \<noun\>**.
```

File: StyleGuide.md [hunk 16]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+
+  **Example:** Stream a Changefeed to Snowflake
+
+- First section describes [prerequisites](#prerequisites). Heading title: **Before you begin** or **Prerequisites**.
```
Comment (ianjevans) [inline StyleGuide.md:329]:
```suggestion
- The first section describes [prerequisites](#prerequisites). Heading title: **Before you begin** or **Prerequisites**.
```

File: StyleGuide.md [hunk 17]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+- Subsequent headings are **Step 1. \<Imperative verb\> a \<noun\>**, **Step 2. \<Imperative verb\> a \<noun\>**, etc., each containing a small ordered list of steps. Within each **Step**, limit the number of steps. Aim for the heuristic maximum of 10 steps.
+- Should be written in a conversational [tone](#language-and-tone), as if it is teaching the user.
```
Comment (ianjevans) [inline StyleGuide.md:331]:
```suggestion
- Tutorials should be written in a conversational [tone](#language-and-tone), as if it is teaching the user.
```

File: StyleGuide.md [hunk 18]
```diff
@@ -147,108 +180,240 @@ For more detail about how to format text, see [Components](#components).
...
+- The instructions should be prescriptive (i.e., tell the user exactly what to do).
+- Should be self-contained. The reader shouldn't have to consult multiple pages to complete the tutorial.
```
Comment (ianjevans) [inline StyleGuide.md:333]:
```suggestion
- Tutorials should be self-contained. The reader shouldn't have to consult multiple pages to complete the tutorial.
```

**Review by ianjevans — APPROVED** [review]
Looks good. I left some suggestions, which can take or leave.

**Review by lnhsingh — COMMENTED** [review]
LGTM. This is a nice write up!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12806#-:-Mx-hv1hDgC4lnOqFwL_:ba4f3k0)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jseldess, @mikeCRL, @rmloveland, and @stbof)

**Review by jseldess — APPROVED** [review]
LGTM. Very nice write-ups!

**Review by ianjevans — APPROVED** [review]

**Review by rmloveland — APPROVED** [review]
LGTM

**Pull request merged**